posthog/api/feature_flag.py

```diff
@@ -121,9 +129,121 @@ def get_queryset(self) -> QuerySet:
         return queryset.order_by("-created_at")
 
     @action(methods=["GET"], detail=False)
-    def for_user(self, request: request.Request, **kwargs):
-        distinct_id = request.GET.get("distinct_id", None)
-        if not distinct_id:
-            raise serializers.ValidationError("Please provide a distinct_id to continue.")
-        flags = get_active_feature_flags(self.team, distinct_id)
-        return Response({"distinct_id": distinct_id, "flags_enabled": flags})
+    def my_flags(self, request: request.Request, **kwargs):
+        if not request.user.is_authenticated:  # for mypy
+            raise exceptions.NotAuthenticated()
+        feature_flags = (
+            FeatureFlag.objects.filter(team=self.team, active=True, deleted=False)
+            .prefetch_related(
+                Prefetch(
+                    "featureflagoverride_set",
+                    queryset=FeatureFlagOverride.objects.filter(user=request.user),
+                    to_attr="my_overrides",
+                )
+            )
+            .order_by("-created_at")
+        )
+        flags = []
+        for feature_flag in feature_flags:
+            my_overrides = cast(List[FeatureFlagOverride], feature_flag.my_overrides)  # type: ignore
+            override = None
+            if len(my_overrides) > 0:
+                override = my_overrides[0]
+
+            value_for_user_without_override: Union[bool, str, None] = feature_flag.distinct_id_matches(
+                request.user.distinct_id
+            )
+            if len(feature_flag.variants) > 0 and value_for_user_without_override:
+                value_for_user_without_override = feature_flag.get_variant_for_distinct_id(request.user.distinct_id)
+
+            flags.append(
+                {
+                    "feature_flag": FeatureFlagSerializer(feature_flag).data,
+                    "value_for_user_without_override": value_for_user_without_override,
+                    "override": FeatureFlagOverrideSerializer(override).data if override else None,
+                }
+            )
+        return Response(flags)
+
+
+class FeatureFlagOverrideSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = FeatureFlagOverride
+        fields = [
+            "id",
+            "feature_flag",
+            "user",
+            "override_value",
+        ]
+
+    _analytics_updated_event_name = "feature flag override updated"
+    _analytics_created_event_name = "feature flag override created"
```

Comment:
It may be kind of confusing down the line to do part of this instrumentation with `AnalyticsDestroyModelMixin`, and part inside the serializer in a bespoke way. Should we just create analogous `AnalyticsCreateModelMixin`/`AnalyticsUpdateModelMixin` to have a plug-n-play solution for everything here?
CC @paolodamico